ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/LayeredPendingTransactions.java

```diff
@@ -0,0 +1,456 @@
+    final Set<Hash> alreadyChecked = new HashSet<>();
+    final AtomicBoolean completed = new AtomicBoolean(false);
+
+    prioritizedTransactions.stream()
+        .takeWhile(unused -> !completed.get())
+        .peek(
+            highPrioPendingTx ->
+                LOG.atDebug()
+                    .setMessage("highPrioPendingTx {}, senderTxs {}")
+                    .addArgument(highPrioPendingTx::toTraceLog)
+                    .addArgument(
+                        () ->
+                            prioritizedTransactions.stream(highPrioPendingTx.getSender())
+                                .map(PendingTransaction::toTraceLog)
+                                .collect(Collectors.joining(", ")))
+                    .log())
+        .forEach(
+            highPrioPendingTx ->
+                prioritizedTransactions.stream(highPrioPendingTx.getSender())
+                    .takeWhile(unused -> !completed.get())
+                    .filter(
+                        candidatePendingTx ->
+                            !alreadyChecked.contains(candidatePendingTx.getHash()))
+                    .filter(
+                        candidatePendingTx ->
+                            candidatePendingTx.getNonce() <= highPrioPendingTx.getNonce())
+                    .forEach(
+                        candidatePendingTx -> {
+                          alreadyChecked.add(candidatePendingTx.getHash());
+                          switch (selector.evaluateTransaction(
+                              candidatePendingTx.getTransaction())) {
+                            case CONTINUE:
+                              LOG.atDebug()
+                                  .setMessage("CONTINUE: Transaction {}")
+                                  .addArgument(candidatePendingTx::toTraceLog)
+                                  .log();
+                              break;
+                            case DELETE_TRANSACTION_AND_CONTINUE:
+                              invalidTransactions.add(candidatePendingTx);
+                              LOG.atDebug()
+                                  .setMessage("DELETE_TRANSACTION_AND_CONTINUE: Transaction {}")
+                                  .addArgument(candidatePendingTx::toTraceLog)
+                                  .log();
+                              logTransactionForReplayDelete(candidatePendingTx);
+                              break;
+                            case COMPLETE_OPERATION:
+                              completed.set(true);
+                              LOG.atDebug()
+                                  .setMessage("COMPLETE_OPERATION: Transaction {}")
+                                  .addArgument(candidatePendingTx::toTraceLog)
+                                  .log();
+                              break;
+                          }
+                        }));
+
+    invalidTransactions.forEach(prioritizedTransactions::invalidate);
```

Comment:
yes what you say could happen, and we could change our idea in future if we collect enough data to state that it is worth to keep these txs around and they are not used as a spam attack vectors, because otherwise could be easy to just fill the pool with high value txs that are not executable because you do not have the balance to pay for them.
So for the moment I want to just keep the same behavior that we already have and invalidate tx if the sender does not have enough balance.